plugins/parsers/json/parser.go

```diff
@@ -78,14 +79,31 @@ func (p *JSONParser) parseObject(metrics []telegraf.Metric, jsonOut map[string]i
 			return nil, err
 		}
 
-		timeStr, ok := f.Fields[p.JSONTimeKey].(string)
-		if !ok {
-			err := fmt.Errorf("time: %v could not be converted to string", f.Fields[p.JSONTimeKey])
-			return nil, err
-		}
-		nTime, err = time.Parse(p.JSONTimeFormat, timeStr)
-		if err != nil {
-			return nil, err
+		// unix: epoch is assumed to be in seconds and unquoted. Can have a decimal part.
+		// unix_ms: epoch is assumed to be in milliseconds and unquoted. Cannot have a decimal part.
+		if strings.EqualFold(p.JSONTimeFormat, "unix") || strings.EqualFold(p.JSONTimeFormat, "unix_ms") {
+			timeEpoch, ok := f.Fields[p.JSONTimeKey].(float64) //Using float64 as that is what the JSONFlattener returns
```

Comment:
I think we should support taking this value from a float or a string, if we take it from a string we should use string processing to split the parts to avoid precision loss (see my other comment).

Reply:
Ok, working on it :thumbsup:  I'll try to support both dot and comma as decimal separators (just in case).

Reply:
What I've seen is that when the JSON field is unquoted, the `JSONFlattener` returns a `float64`, so there could be already an unavoidable precision loss, specially if the value has more than a few decimal positions. This can be avoided using string processing as you suggested, so I think we should clarify in the docs that if precision accuracy is required, the value must be sent as string. On the other hand, if that is not required or if not using decimals at all (like in `unix_ms` option), the JSON value can come quoted or unquoted. What do you think?

Reply:
A note in the docs about potential precision loss would be a nice touch.  Later on, we can explore updating the JSONFlattener code to use `json.Number`.

Reply:
Great! Just commited the changes!